packages/net/writebuffer.pony

```diff
@@ -0,0 +1,312 @@
+
+  fun ref f64_le(data: F64): WriteBuffer^ =>
+    """
+    Write an F64 to the buffer in little-endian byte order.
+    """
+    u64_le(data.bits())
+
+  fun ref f64_be(data: F64): WriteBuffer^ =>
+    """
+    Write an F64 to the buffer in big-endian byte order.
+    """
+    u64_be(data.bits())
+
+  fun ref u128_le(data: U128): WriteBuffer^ =>
+    """
+    Write a U128 to the buffer in little-endian byte order.
+    """
+    _check(16)
+    _byte(data.u8())
+    _byte((data >> 8).u8())
+    _byte((data >> 16).u8())
+    _byte((data >> 24).u8())
+    _byte((data >> 32).u8())
+    _byte((data >> 40).u8())
+    _byte((data >> 48).u8())
+    _byte((data >> 56).u8())
+    _byte((data >> 64).u8())
+    _byte((data >> 72).u8())
+    _byte((data >> 80).u8())
+    _byte((data >> 88).u8())
+    _byte((data >> 96).u8())
+    _byte((data >> 104).u8())
+    _byte((data >> 112).u8())
+    _byte((data >> 120).u8())
+    this
+
+  fun ref u128_be(data: U128): WriteBuffer^ =>
+    """
+    Write a U128 to the buffer in big-endian byte order.
+    """
+    _check(16)
+    _byte((data >> 120).u8())
+    _byte((data >> 112).u8())
+    _byte((data >> 104).u8())
+    _byte((data >> 96).u8())
+    _byte((data >> 88).u8())
+    _byte((data >> 80).u8())
+    _byte((data >> 72).u8())
+    _byte((data >> 64).u8())
+    _byte((data >> 56).u8())
+    _byte((data >> 48).u8())
+    _byte((data >> 40).u8())
+    _byte((data >> 32).u8())
+    _byte((data >> 24).u8())
+    _byte((data >> 16).u8())
+    _byte((data >> 8).u8())
+    _byte(data.u8())
+    this
+
+  fun ref i128_le(data: I128): WriteBuffer^ =>
+    """
+    Write an I128 to the buffer in little-endian byte order.
+    """
+    u128_le(data.u128())
+
+  fun ref i128_be(data: I128): WriteBuffer^ =>
+    """
+    Write an I128 to the buffer in big-endian byte order.
+    """
+    u128_be(data.u128())
+
+  fun ref write(data: ByteSeq): WriteBuffer^ =>
+    """
+    Write a ByteSeq to the buffer.
+    """
+    _append_current()
+    _chunks.push(data)
+    _size = _size + data.size()
+    this
+
+  fun ref writev(data: ByteSeqIter): WriteBuffer^ =>
+    """
+    Write ByteSeqs to the buffer.
+    """
+    _append_current()
+    for chunk in data.values() do
+      _chunks.push(chunk)
+      _size = _size + chunk.size()
+    end
+    this
+
+  fun ref done(): Array[ByteSeq] iso^ =>
+    """
+    Return an array of buffered ByteSeqs and reset the WriteBuffer's buffer.
+    """
+    _append_current()
+    _size = 0
+    _chunks = recover Array[ByteSeq] end
+
+  fun ref _append_current() =>
+    if _offset > 0 then
+      _current.truncate(_offset)
+      _offset = 0
+      _chunks.push(_current = recover Array[U8] end)
+    end
+
+  fun ref _check(size': USize) =>
+    if (_current.size() - _offset) < size' then
+      _current.undefined(_offset + size')
+    end
+
+  fun ref _byte(data: U8) =>
+    try
+      _current(_offset) = data
+      _offset = _offset + 1
+      _size = _size + 1
+    end
```

Comment:
I'll fix that.

> On May 12, 2016, at 11:27 PM, Joe Eli McIlvain <notifications@github.com> wrote:
> 
> In packages/net/writebuffer.pony:
> 
> > +      _current.truncate(_offset)
> > +      _offset = 0
> > +      _chunks.push(_current = recover Array[U8] end)
> > +    end
> > +
> > +  fun ref _check(size': USize) =>
> > +    if (_current.size() - _offset) < size' then
> > +      _current.undefined(_offset + size')
> > +    end
> > +
> > +  fun ref _byte(data: U8) =>
> > +    try
> > +      _current(_offset) = data
> > +      _offset = _offset + 1
> > +      _size = _size + 1
> > +    end
> Formatting nitpick... can you make sure there is a single newline at the end of this file to get rid of the GitHub "no newline at end of file" warning?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
>